client/castApp/src/castinit.c

```diff
@@ -200,6 +200,84 @@ static void addReccasterEnvVarsCallFunc(const iocshArgBuf *args)
     addReccasterEnvVars(&thecaster, args[0].aval.ac, args[0].aval.av);
 }
 
+void addReccasterExcludePattern(caster_t* self, int argc, char **argv) {
+    size_t i;
+    int ret = 0;
+    argv++; argc--; /* skip function arg */
+    if(argc < 1) {
+        errlogSevPrintf(errlogMinor, "At least one argument expected for addReccasterExcludePattern\n");
+        return;
+    }
+    epicsMutexMustLock(self->lock);
+    if(self->shutdown) {
+        /* shutdown in progress, silent no-op */
+        epicsMutexUnlock(self->lock);
+        return;
+    }
+    // error if called after iocInit()
+    if(self->current != casterStateInit) {
+        errlogSevPrintf(errlogMinor, "addReccasterExcludePattern called after iocInit() when reccaster might already be connected. Not supported\n");
+        epicsMutexUnlock(self->lock);
+        return;
+    }
+
+    for (i = 0; i < argc; i++) {
+        if (argv[i] == NULL) {
+            errlogSevPrintf(errlogMinor, "Arg is NULL for addReccasterExcludePattern\n");
+        }
```

Comment:
Honest question here: Can this actually happen? I would assume that the `argv`s will never be null, but I am not certain...